### PR TITLE
[BEAM-3611] Split KafkaIO.java in to smaller files

### PR DESCRIPTION
**Pull request by rangadi**
KafkaIO.java has grown to be 2.5 lines and contains multiple independent components in single class : Source, Reader, Writer DoFn, ExactlyOnceWriter etc. I have split it into multiple files as following. There are **no functional changes** and all are package private.

-  KafkaIO.java : Contains top level API used by users : top level Read() and Write() transforms.
-  KafkaUnboundedSource.java : Implements UnboundedSource API
-  KafkaUnboundedReader.java : Implements UnboundedReader API
-  KafkaWriter.java : Writer DoFn used in KafkaIO sink
-  KafkaExactlyOnceSink : Sink transform used when exactly-once semantics are enabled.

### CODE REVIEW

**Comment by rangadi** [comment]
+R: @XuMingmin @jkff

File: sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaRecordCoder.java [hunk 1]
```diff
@@ -50,37 +49,22 @@ public KafkaRecordCoder(Coder<K> keyCoder, Coder<V> valueCoder) {
   }
 
   @Override
-  public void encode(KafkaRecord<K, V> value, OutputStream outStream)
-      throws CoderException, IOException {
-    encode(value, outStream, Context.NESTED);
+  public void encode(KafkaRecord<K, V> value, OutputStream outStream) throws IOException {
+    stringCoder.encode(value.getTopic(), outStream);
```
Comment (jkff) [inline sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaRecordCoder.java]: Why this change? Is it also a semantic no-op? (i.e. update-compatible)

Comment (rangadi) [reply sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaRecordCoder.java]: Looks like context is not used anymore and API is deprecated. Must be a no-op. May be SDK runtime has stopped differentiating 'outer' and 'nested'. I noticed deprecation warnings since couple of weeks back and thought it might be safer to try in 2.4. Deprecation messages in JavaDoc for `Context` does not talk about why it is removed and does not warn about any unintended consequences. Important coders `KvCoder` seem to ignore it as well..

**Comment by rangadi** [comment]
@XuMingmin replied to @jkff question above. I think it is safe and recommended. I can remove the change to coder if that is better.

**Comment by jkff** [comment]
retest this please
